.github/scripts/transform_dbt_test_results.py

```diff
@@ -0,0 +1,1082 @@
+        run_dt_str = run_results["metadata"]["generated_at"]
+        run_dt = datetime.datetime.strptime(
+            run_dt_str, "%Y-%m-%dT%H:%M:%S.%fZ"
+        )
+        run_date = run_dt.strftime("%Y-%m-%d")
+        elapsed_time = run_results["elapsed_time"]
+
+        # Extract dbt vars
+        run_vars = run_results["args"]["vars"]
+        var_year_start = run_vars.get("test_qc_year_start")
+        var_year_end = run_vars.get("test_qc_year_start")
+
+        # If dbt vars weren't set on the command line, the defaults won't exist
+        # in run_results.json, so we have to parse them from the dbt project
+        # config
+        if not var_year_start or not var_year_end:
+            with open("dbt_project.yml") as project_fobj:
+                project = yaml.safe_load(project_fobj)
+            var_year_start = (
+                var_year_start or project["vars"]["test_qc_year_start"]
+            )
+            var_year_end = var_year_end or project["vars"]["test_qc_year_end"]
+
+        return cls(
+            run_id=run_id,
+            run_date=run_date,
+            elapsed_time=elapsed_time,
+            var_year_start=var_year_start,
+            var_year_end=var_year_end,
+        )
+
+
+@dataclasses.dataclass
+class TestRunResultMetadata:
+    """Metadata object storing information about test results in a run."""
+
+    run_id: str
+    test_name: str
+    table_name: str
+    category: str
+    description: str
+    township_code: typing.Optional[str]
+    status: str  # Serialize Status enum to str for output to parquet
+    elapsed_time: decimal.Decimal
+    num_failing_rows: int
+
+    @classmethod
+    def create_list(
+        cls,
+        test_categories: typing.List[TestCategory],
+        run_results_filepath: str,
+    ) -> typing.List["TestRunResultMetadata"]:
+        """Generate a list of TestRunMetadata object from a list of
+        TestCategory objects representing the categories in the run and a
+        filepath to a run_results.json file."""
+        with open(run_results_filepath) as run_results_fobj:
+            run_results = json.load(run_results_fobj)
+
+        run_id = run_results["metadata"]["invocation_id"]
+
+        return [
+            TestRunResultMetadata(
+                run_id=run_id,
+                test_name=township_result.name,
+                table_name=township_result.table_name,
+                category=test_category.category,
+                description=township_result.description,
+                township_code=township_result.township_code,
+                status=township_result.status.value,
+                elapsed_time=township_result.elapsed_time,
+                num_failing_rows=len(township_result.failing_rows),
+            )
+            for test_category in test_categories
+            for test_result in test_category.test_results
+            for township_result in test_result.split_by_township()
+        ]
+
+
+def get_test_cache_path(
```

Comment:
`get_test_cache_path`, `get_test_categories_from_file`, `save_test_categories_to_file`, and `get_test_categories_from_athena` are basically unchanged since the last version of this script; the only changes are to symbol names, and using the new `TestCategory` interface for (de)serialization.